libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -54,66 +54,83 @@ string IRLocalVariable::setToZero() const
 IRStorageItem::IRStorageItem(
 	IRGenerationContext& _context,
 	VariableDeclaration const& _varDecl
-):
-	IRLValue(_context, _varDecl.annotation().type)
+)
+:IRStorageItem(
+	_context,
+	*_varDecl.annotation().type,
+	_context.storageLocationOfVariable(_varDecl)
+)
+{ }
+
+IRStorageItem::IRStorageItem(
+	IRGenerationContext& _context,
+	Type const& _type,
+	std::pair<u256, unsigned> slot_offset
+)
+:	IRLValue(_context, &_type),
+	m_slot(toCompactHexWithPrefix(slot_offset.first)),
+	m_offset(slot_offset.second)
 {
-	u256 slot;
-	unsigned offset;
-	std::tie(slot, offset) = _context.storageLocationOfVariable(_varDecl);
-	m_slot = toCompactHexWithPrefix(slot);
-	m_offset = offset;
 }
 
 IRStorageItem::IRStorageItem(
 	IRGenerationContext& _context,
 	string _slot,
-	unsigned _offset,
+	boost::variant<string, unsigned> const& _offset,
 	Type const& _type
 ):
 	IRLValue(_context, &_type),
 	m_slot(move(_slot)),
-	m_offset(_offset)
+	m_offset(std::move(_offset))
```

Comment:
Yeah - this ``std::move`` here doesn't actually move anything right now. The canonical way to do this is to change the argument type to ``boost::variant<string, unsigned> _offset`` and then to use ``std::move`` here as well as when calling this.